### ISSUE TITLE
Conditionalize MaxRecvMessageSize

### ISSUE THREAD

**Issue by vibe** [crossplane-contrib/function-auto-ready#40]
There was a need to bump `MaxRecvMessageSize` across all our functions recently with the upgrade to the new v1 sdk, this included `function-auto-ready`. 

Just dropping an issue to track this, and get some feedback about where the best place to implement this is. 

Simple fix is to update the fn.Serve with `function.MaxRecvMessageSize(c.MaxGRPCMessageSize)`.

Could either happen at each function level, or `function-sdk-go` template.

**Comment by jbw976** [comment]
Would you be able to add a bit more detail on what the problem is here @vibe? I don't fully understand the scope and impact just yet 🤔

**Comment by vibe** [comment]
> Would you be able to add a bit more detail on what the problem is here @vibe? I don't fully understand the scope and impact just yet 🤔

Sure @jbw976. 

Essentially, in Go's gRPC library, the default maximum received message size is 4MB. So messages sent between composition functions need to be remain under this limit, since this includes the observed composite resource and the desiresd resources, this message size can increase as more resources are being generated by a composition pipeline. 

A few months ago we hit the 4MB limit (our composition, provisions large amounts of EC2 with custom user data), we then refactored the user data so that the total size would fall below 4MB.

Fast forward to last week: We bumped our `function-sdk-go` to 0.3.0 and the message size between our composition increased resulting in an error along the lines of `received message larger than max (420000 vs. 4194304)"`. Rolling back the `function-sdk-go` lowered the message size. 

We were easily able to update our own function's `MaxRecvMessageSize` but not `function-auto-ready` so we had to fork it and make the change.


It makes sense [function-template-go](https://github.com/crossplane/function-template-go) is updated, along with any other functions that may need this change. 

I can work on a PR soon.

**Comment by fernandezcuesta** [comment]
Yeah here we faced the same issue, which surfaced with `function-auto-ready`, since that's the last in the pipeline and we've quite a fat composition

**Comment by fnicolelli-ls** [comment]
Hi team, same issue here we hit the limit in function-auto-ready since we have a quite big composition that unfortunately cannot be split.
It would be awesome have the possibility to set the max size.
Thanks!

**Comment by vibe** [comment]
Sorry, finally go around to this.
